src/data/finsupp/ne_locus.lean

```diff
@@ -88,35 +88,44 @@ by { ext, simpa only [mem_ne_locus] using hF.ne_iff }
 
 end ne_locus_and_maps
 
-section cancel_and_group
 variables [decidable_eq N]
 
-@[simp] lemma add_ne_locus_add_eq_left [add_left_cancel_monoid N] (f g h : α →₀ N) :
-  (f + g).ne_locus (f + h) = g.ne_locus h  :=
+@[simp] lemma ne_locus_add_left [add_left_cancel_monoid N] (f g h : α →₀ N) :
+  (f + g).ne_locus (f + h) = g.ne_locus h :=
 zip_with_ne_locus_eq_left _ _ _ _ add_right_injective
 
-@[simp] lemma add_ne_locus_add_eq_right [add_right_cancel_monoid N] (f g h : α →₀ N) :
-  (f + h).ne_locus (g + h) = f.ne_locus g  :=
+@[simp] lemma ne_locus_add_right [add_right_cancel_monoid N] (f g h : α →₀ N) :
+  (f + h).ne_locus (g + h) = f.ne_locus g :=
 zip_with_ne_locus_eq_right _ _ _ _ add_left_injective
 
-@[simp] lemma neg_ne_locus_neg [add_group N] (f g : α →₀ N) : (- f).ne_locus (- g) = f.ne_locus g :=
+section add_group
+variables [add_group N] (f f₁ f₂ g g₁ g₂ : α →₀ N)
+
+@[simp] lemma ne_locus_neg_neg : ne_locus (-f) (-g) = f.ne_locus g :=
 map_range_ne_locus_eq _ _ neg_zero neg_injective
 
-lemma ne_locus_neg [add_group N] (f g : α →₀ N) : (- f).ne_locus g = f.ne_locus (- g) :=
-by rw [← neg_ne_locus_neg _ _, neg_neg]
+lemma ne_locus_neg : ne_locus (-f) g = f.ne_locus (-g) := by rw [←ne_locus_neg_neg, neg_neg]
+
+@[simp] lemma support_sub_eq_ne_locus (f g : α →₀ N) : (f - g).support = f.ne_locus g :=
+by rw [←ne_locus_add_right _ _ (-g), add_right_neg, ne_locus_zero_right, sub_eq_add_neg]
```

Comment:
I'm not sure why you would like to change this lemma? The counterpart [dist_eq_norm](https://leanprover-community.github.io/mathlib_docs/analysis/normed/group/basic.html#dist_eq_norm) of this lemma is in the opposite direction and is not a simp lemma, which matches the original lemma but not your new version.

Reply:
i guess that makes sense, yeah.